src/vs/editor/contrib/rtv/RTVSynthService.ts

```diff
@@ -0,0 +1,327 @@
+import { TableElement, isHtmlEscape } from 'vs/editor/contrib/rtv/RTVUtils';
```

Comment:
Could we try to store the entire box state here in the model? That way, we could implement an "Undo" that moves the user back into the box by simply copying the last state of the box before exiting. Then if the user Ctrl+zs to that point, instead of creating a new model, we can restore the old one and (in theory) that restores the entire state (table content, which lines were highlighted, the cursor position, etc.)

Reply:
Yes! Now that we keep all the necessary info in `RTVSynthModel`, I think it is doable. Added to my future TODOs :)